packages/puppeteer-core/src/common/bidi/BrowsingContext.ts

```diff
@@ -51,6 +53,54 @@ const lifeCycleToReadinessState = new Map<
   ['domcontentloaded', 'interactive'],
 ]);
 
+/**
+ * @internal
+ */
+export class CDPSessionWrapper extends EventEmitter implements CDPSession {
```

Comment:
Maybe this class is should live in the Connection file?

Reply:
Not sure, my thinking was is that it wraps the context and is only accessible from the context so we could keep it here?

Reply:
Let's keep it as is.